### PR TITLE
Make consuming take time

### PR DESCRIPTION
**Pull request by Ramza13**
#### Summary
SUMMARY: Features "Makes consuming items take time"

#### Purpose of change
First part of #36972
Makes consuming items take time

#### Describe the solution
Added a new activity actor for consuming which uses the formula in the linked PR for times.  Medicine/drugs weren't mentioned in the linked PR so for right now they all take 30 seconds. 

#### Describe alternatives you've considered
Completing #36972 in this PR, instead decided to break it up and implement the rest of it in a future PR. 

#### Testing
Spawned crack, crack pipe,bandages,antiseptic, can of soda, hamburger, wrapper of protein bar.  Damaged limbs, used healing items, smoked crack, ate and drank the rest.

### CODE REVIEW

File: src/activity_actor.cpp [hunk 1]
```diff
@@ -409,12 +412,77 @@ std::unique_ptr<activity_actor> open_gate_activity_actor::deserialize( JsonIn &j
...
+    }
+
+    act.moves_total = time;
+    act.moves_left = time;
+}
+
+void consume_activity_actor::do_turn( player_activity &act, Character & )
+{
+    act.moves_left = act.moves_left - 1;
+    g->u.moves -= 100;
```
Comment (ifreund) [inline src/activity_actor.cpp:440]: If you set the activity type to be based on time or speed in the json, the `player_activity` class will tick these down for you. I think speed is probably more appropriate here as a faster player will likely eat marginally faster.

**Comment by Mecares** [comment]
Great addition would it be hard to influence the time to consume something by traits, like make a character with rigid tablemanners or gourmand take longer to eat something?

File: src/activity_actor.cpp [hunk 2]
```diff
@@ -409,12 +412,71 @@ std::unique_ptr<activity_actor> open_gate_activity_actor::deserialize( JsonIn &j
     return actor.clone();
 }
 
+void consume_activity_actor::start( player_activity &act, Character &who )
+{
+    item &comest = who.get_consumable_from( *loc.get_item() );
+
+    const int charges = std::max( comest.charges, 1 );
+    int volume = units::to_milliliter( comest.volume() ) / charges;
+    int time = 0;
```
Comment (KorGgenT) [inline src/activity_actor.cpp:421]: i suggest making this a time_duration for ease of reading the code, and for maintenance

File: src/activity_actor.cpp [hunk 3]
```diff
@@ -409,12 +412,71 @@ std::unique_ptr<activity_actor> open_gate_activity_actor::deserialize( JsonIn &j
     return actor.clone();
 }
 
+void consume_activity_actor::start( player_activity &act, Character &who )
+{
+    item &comest = who.get_consumable_from( *loc.get_item() );
```
Comment (KorGgenT) [inline src/activity_actor.cpp:417]: get_consumable_from is now obsolete. *loc is the same as *loc.get_item()

File: src/activity_actor.cpp [hunk 4]
```diff
@@ -409,12 +412,71 @@ std::unique_ptr<activity_actor> open_gate_activity_actor::deserialize( JsonIn &j
...
+
+    const int charges = std::max( comest.charges, 1 );
+    int volume = units::to_milliliter( comest.volume() ) / charges;
```
Comment (KorGgenT) [inline src/activity_actor.cpp:420]: comest.volume() is giving you the volume of the whole stack, don't you just want the volume of one from here?

Comment (Ramza13) [reply src/activity_actor.cpp:420]: I'm copying the logic used for displaying the amount of food you are eating in the consume menu.

File: src/activity_actor.cpp [hunk 5]
```diff
@@ -409,12 +412,71 @@ std::unique_ptr<activity_actor> open_gate_activity_actor::deserialize( JsonIn &j
...
+        debugmsg( "Consumed something that was not food, drink or medicine/drugs" );
+    }
+    time *= 100; //Convert from seconds to moves
+    act.moves_total = time;
+    act.moves_left = time;
+}
+
+void consume_activity_actor::finish( player_activity &act, Character & )
+{
+    item *it = loc.get_item();
```
Comment (KorGgenT) [inline src/activity_actor.cpp:439]: what's the point of this here? you don't even use it once other than dereferencing it, and ou can just as easily dereference an item_location `*loc`

**Comment by Ramza13** [comment]
> 
> 
> Great addition would it be hard to influence the time to consume something by traits, like make a character with rigid tablemanners or gourmand take longer to eat something and maybe some mutations make one eat faster?

I plan to do that in a followup PR when I do most of the rest of the linked PR.( I don't think I will add the optional eating speed stuff as that seems a little UI heavy for little gain).

**Pull request closed without merging**